Python/import.c

```diff
@@ -122,6 +122,13 @@ _PyImport_ReleaseLock(PyInterpreterState *interp)
     _PyRecursiveMutex_Unlock(&IMPORT_LOCK(interp));
 }
 
+void
+_PyImport_ReInitLock(PyInterpreterState *interp)
+{
+    // gh-126688: Thread id may change after fork() on some operating systems.
+    IMPORT_LOCK(interp).thread = PyThread_get_thread_ident_ex();
```

Comment:
Wouldn't this lead to deadlocks if some other thread happened to be holding the import lock when we forked?  It seems like the child process needs to know the thread ID from which it forked.

Reply:
We acquire the import lock before forking so that no other thread holds it:

https://github.com/python/cpython/blob/09c240f20c47db126ad7e162df41e5c2596962d4/Modules/posixmodule.c#L618-L624

Reply:
Hmm, looks like we always acquire the import lock in `PyOS_BeforeFork()`.  It should be fine then.